### PR TITLE
Converted CLI to MVC

### PR DESCRIPTION
**Pull request by JonathanMathBoi**
Split CLI functionality into CLIController and CLIView to use the MVC design pattern. Save Load functionality still needs to be added.

### CODE REVIEW

File: app/src/main/java/xterminators/spellingbee/cli/CLIView.java [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+     * @param foundWords The list of found words
+     */
+    public void showFoundWords(List<String> foundWords) {
+        if (foundWords.isEmpty()) {
+            System.out.println("You have not found any words yet.");
+        } else if (foundWords.size() == 1) {
+            System.out.println("You have found 1 word:");
+        } else {
+            System.out.println("You have found " + foundWords.size() + " words:");
+        }
```
Comment (ljvance) [inline app/src/main/java/xterminators/spellingbee/cli/CLIView.java:29]: Why is there a different print statement when they have found 1 word and > 1 word?

File: app/src/main/java/xterminators/spellingbee/cli/CLIView.java [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            System.out.println(
+                "Current Rank: " + rank.getRankName() + " - " +
+                earnedPoints + " point\n"
+            );
+        } else {
+            System.out.println(
+                "Current Rank: " + rank.getRankName() + " - " +
+                earnedPoints + " points\n"
+            );
+        }
```
Comment (ljvance) [inline app/src/main/java/xterminators/spellingbee/cli/CLIView.java:140]: Again, why different prints for 1 and >1 when they do basically the same thing.

**Review by ljvance — APPROVED** [review]
Some Minor comments, but they shouldn't have any impact on functionality

File: app/src/main/java/xterminators/spellingbee/cli/CLIView.java (hunk 1 above)
Comment (JonathanMathBoi) [reply app/src/main/java/xterminators/spellingbee/cli/CLIView.java:29]: Singular vs plural. So "found 1 word" vs "found n word**s**".

**Review by ljvance — APPROVED** [review]
looks good

**Pull request merged**